src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsWorkItemEmbededLinkEnricher.cs

```diff
@@ -0,0 +1,145 @@
+                            var workItemId = workItemLinkMatch.Groups["id"].Value;
+                            Log.LogDebug($"{logTypeName}: Source work item {workItemId} mention link traced on field {field.Name} on target work item {targetWorkItem.Id}.");
+                            var sourceLinkWi = Engine.Source.WorkItems.GetWorkItem(workItemId);
+                            if (sourceLinkWi != null)
+                            {
+                                var linkWI = Engine.Target.WorkItems.FindReflectedWorkItemByReflectedWorkItemId(sourceLinkWi);
+                                if (linkWI != null)
+                                {
+                                    var replaceValue = anchorTagMatch.Value.Replace(workItemId, linkWI.Id);
+                                    field.Value = field.Value.ToString().Replace(anchorTagMatch.Value, replaceValue);
+                                    Log.LogInformation($"{logTypeName}: Source work item {workItemId} mention link was successfully replaced with target work item {linkWI.Id} mention link on field {field.Name} on target work item {targetWorkItem.Id}.");
+                                }
+                                else
+                                {
+                                    Log.LogError($"{logTypeName}: Matching target work item mention link for source work item {workItemId} mention link on field {field.Name} on target work item {targetWorkItem.Id} was not found on the target collection.");
+                                }
+                            }
+                            else
+                            {
+                                Log.LogInformation($"{logTypeName}: [SKIP] Source work item {workItemId} mention link on field {field.Name} on target work item {targetWorkItem.Id} was not found on the source collection.");
+                            }
+                        }
+                        else if ((href.StartsWith("mailto:") || href.StartsWith("#")) && value.StartsWith("@"))
+                        {
+                            var displayName = value.Substring(1);
+                            Log.LogDebug($"{logTypeName}: User identity {displayName} mention traced on field {field.Name} on target work item {targetWorkItem.Id}.");
+                            var identity = _targetTeamFoundationIdentities.Value.FirstOrDefault(i => i.DisplayName == displayName);
+                            if (identity != null)
+                            {
+                                var replaceValue = anchorTagMatch.Value.Replace(href, "#").Replace(version, $"data-vss-mention=\"version:2.0,{identity.TeamFoundationId}\"");
+                                field.Value = field.Value.ToString().Replace(anchorTagMatch.Value, replaceValue);
+                                Log.LogInformation($"{logTypeName}: User identity {displayName} mention was successfully matched on field {field.Name} on target work item {targetWorkItem.Id}.");
```

Comment:
All Logging should be in the format `"My message that contains {item} and {item2}", item, itesm2`! Do not use `$"My message that contains {item} and {item2}"` to pass text into the log strings as this disables [Serilog](https://serilog.net/)'s ability to pass that data as telemetry to Application Insights and for log highlighting.

Reply:
Updated..